### ISSUE TITLE
QoL change: Move examples from HTML to Wiki

### ISSUE THREAD

**Issue by realkarmakun** [kwestlund/cooldb#3]
I was trying to figure everything out through javadocs but you do have examples. They just hidden away in HTML file. It would be awesome to setup Wiki page here on Github I even took liberty of converting HTML to Markdown so you can add it right away (if that's okay with you of course):

[Examples.md](https://github.com/kwestlund/cooldb/files/8700920/Examples.md)

**Comment by kwestlund** [comment]
Hi realkarmakun,

This was mostly a research project from many years ago—if you don’t mind me asking, what are you hoping to do with it?

Ken

On May 16, 2022, at 10:26 AM, realkarmakun ***@***.******@***.***>> wrote:



I was trying to figure everything out through javadocs but you do have examples. They just hidden away in HTML file. It would be awesome to setup Wiki page here on Github I even took liberty of converting HTML to Markdown so you can add it right away (if that's okay with you of course):

Examples.md<https://github.com/kwestlund/cooldb/files/8700920/Examples.md>

**Comment by realkarmakun** [comment]
@kwestlund I have a project for Minecraft in mind that would synchronize in-game world changes between multiple Minecraft servers. Right now only ~100 people can play at one Minecraft server (it can be more but there is a lot of restrictions like world size and etc.). I was hoping to make it seamlessly synchronize and load-balance players so for a player it would look like they are playing "on one big server". Right now I'm at early stage where I need reliable embedded WAL complaint database that would keep block changes for future synchronization.

TL;DR trying to scale Minecraft game server horizontally :)

This lib looked solid to me and very much state-of-art, just that API is maybe at early stages. Also I need a Java library so this pure java implementation looked "cool" 😆 
But I can look elsewhere if you don't plan on maintaining this. It's not like I need much anyway. Maybe I'll keep using it without bothering you if you want

**Comment by kwestlund** [comment]
Sounds like a fun project.

I am not actively maintaining cooldb.

I wrote it over many years, starting about 25 years ago, as mainly a research project.

I’d be happy to look into any issues you find or entertain Pull Requests if you still want to try it, but can’t guarantee quick turnaround.

Otherwise, you might have a look at MapDB, which seems well tested, but I’m not sure how well-maintained that is either.

https://github.com/jankotek/mapdb/

Then there are the LevelDB (Google) and derivative systems like RocksDB (Facebook) that have bigger backing. Here’s a java port of LevelDB:

https://github.com/dain/leveldb

Good luck!

On May 16, 2022, at 12:01 PM, realkarmakun ***@***.******@***.***>> wrote:



@kwestlund<https://github.com/kwestlund> I have a project for Minecraft in mind that would synchronize in-game world changes between multiple Minecraft servers. Right now only ~100 people can play at one Minecraft server (it can be more but there is a lot of restrictions like world size and etc.). I was hoping to make it seamlessly synchronize and load-balance players so for a player it would look like they are playing "on one big server". Right now I'm at early stage where I need reliable embedded WAL complaint database that would keep block changes for future synchronization.

TL;DR trying to scale Minecraft game server horizontally :)

This lib looked solid to me and very much state-of-art, just that API is maybe at early stages. Also I need a Java library so this pure java implementation looked "cool" 😆
But I can look elsewhere if you don't plan on maintaining this

**Comment by realkarmakun** [comment]
Thank you for the heads up! ❤️ I'll look into those variants. I'll be closing this now then :)
